src/pages/index.js

```diff
@@ -1,83 +1,191 @@
+    "Content-Type": "application/json",
+  },
+});
 
-//Selecting buttons
+//Selecting elements
 const buttonEdit = document.querySelector(".profile__button_type_edit");
 const buttonAdd = document.querySelector(".profile__button_type_add");
+const avatarEdit = document.querySelector(".image__container");
+
+//Create sectionCard class
+const sectionCard = new Section({ renderer: createCard }, ".cards__list");
+
+//Create UserInfo class
+const userInfo = new UserInfo(userSettings);
 
 //Create popups classes and set event listeners
 const popupImage = new PopupWithImage("#image-modal-id");
 const popupEdit = new PopupWithForm("#edit-modal-id", handleEditSumbit);
 const popupAdd = new PopupWithForm("#add-modal-id", handleAddSumbit);
-
-//Create card section class
-const sectionCard = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
+const popupAvatar = new PopupWithForm("#avatar-modal-id", handleAvatarSubmit);
+const popupDelete = new PopupDelete("#delete-modal-id", handleDeleteSubmit);
 
 //Create validators
 const formEditValidator = new FormValidator(validationSettings, popupEdit.form);
 const formAddValidator = new FormValidator(validationSettings, popupAdd.form);
+const formAvatarValidator = new FormValidator(
+  validationSettings,
+  popupAvatar.form
+);
 
 //Handlers for classes
 function createCard(item) {
-  const cardElement = new Card(item, "#card__template", handleImageClick);
+  const cardElement = new Card(
+    item,
+    "#card__template",
+    handleImageClick,
+    handleDeleteClick,
+    toggleServerLike
+  );
   return cardElement.renderCard();
 }
 
 function handleImageClick(data) {
   popupImage.open(data);
 }
 
+function handleSubmits(request, popup, loadingText = "Saving...") {
+  popup.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popup.renderLoading(false);
+    });
+}
+
 function handleEditSumbit(obj) {
-  userInfo.setUserInfo(obj);
-  popupEdit.close();
-  formEditValidator.disableButton();
+  function makeRequest() {
+    return api.updateCurrentUser(obj).then((data) => {
+      userInfo.setUserInfo(data);
+      formEditValidator.disableButton();
+    });
+  }
+
+  handleSubmits(makeRequest, popupEdit);
 }
 
 function handleAddSumbit(obj) {
-  sectionCard.addItem(createCard(obj));
-  popupAdd.close();
-  popupAdd.form.reset();
-  formAddValidator.disableButton();
+  function makeRequest() {
+    return api.postCard(obj).then((data) => {
+      sectionCard.addItem(createCard(data), "prepend");
+      popupAdd.form.reset();
+      formAddValidator.disableButton();
+    });
+  }
+
+  handleSubmits(makeRequest, popupAdd);
+}
+
+function handleDeleteSubmit(thisCard) {
+  function makeRequest() {
+    return api.deleteCard(thisCard.id).then(() => {
+      thisCard.deleteCard();
+    });
+  }
+
+  handleSubmits(makeRequest, popupDelete);
```

Comment:
One little thins is left:

* When you delete a card, the button should displayed `Deleting...`


```js
  handleSubmits(makeRequest, popupDelete, "Deleting...");


```

Reply:
I may have missed it, but from what I see this in not in the brief.
It's ok, hope the project will be accepted now that I fixed it

Reply:
> I may have missed it, but from what I see this in not in the brief.

You are right, but if change the confirmation button text, then it should contain the right text)


You've done a really cool job. You should be proud of your progress!